odl/discr/tensor_ops.py

```diff
@@ -669,6 +669,67 @@ def adjoint(self):
                               weight=self.weights)
 
 
+# TODO: Optimize this to an optimized operator on its own.
+class PointwiseSum(PointwiseInner):
+
+    """Take the point-wise sum of a vector field.
+
+    This operator takes the (weighted) sum
+
+        ``sum(F(x)) = [ sum_j( w_j * F_j(x) ) ]
+
+    where ``F`` is a vector field. This implies that
+    the `Operator.domain` is a power space of a discretized function
+    space. For example, if ``X`` is a `DiscreteLp` space, then
+    ``ProductSpace(X, d)`` is a valid domain for any positive integer
+    ``d``.
+    """
+
+    def __init__(self, vfspace, weight=None):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        vfspace : `ProductSpace`
+            Space of vector fields on which the operator acts.
+            It has to be a product space of identical spaces, i.e. a
+            power space.
+        weight : `array-like` or float, optional
```

Comment:
I went by simply removing the positivity constraint. It is not really a strict constraint anyway, everything works with negative values.